nevergrad/optimization/recaster.py

```diff
@@ -222,12 +205,108 @@ def __del__(self) -> None:
 
 
 class SequentialRecastOptimizer(RecastOptimizer):
-    """Recast Optimizer which cannot deal with parallelization"""
+    """Recast Optimizer which cannot deal with parallelization
+
+    There can only be one worker. Each ask must be followed by
+    a tell.
+
+    A simple usage is that you have a library which can minimize
+    a function which returns a scalar.
+    Just make an optimizer inheriting from this class, and inplement
+    get_optimization_function to return a callable which runs the
+    optimization, taking the objective as its only parameter. The
+    callable must not have any references to the optimizer itself.
+    (This avoids a reference cycle between the background thread and
+    the optimizer, aiding cleanup.) It can have a weakref though.
+
+    If you want your optimizer to be picklable, we have to store
+    every candidate during optimization, which may use a lot
+    of memory. This lets us replay the optimization when
+    unpickling. To enable this:
+        - set enable_pickling to True in __init__.
+        - The optimization must be reproducible, asking for the same
+          candidates every time. If you need a seed from nevergrad's
+          generator, you can't necessarily generate this again after
+          unpickling. One solution is to store it in self, if it is
+          not there yet, in the body of get_optimization_function.
+
+          As in general in nevergrad, do not set the seed from the
+          RNG in your own __init__ because it will cause surprises
+          to anyone re-seeding your parametrization after init.
+    """
 
     # pylint: disable=abstract-method
 
     no_parallelization = True
 
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: tp.Optional[int],
+        num_workers: int = 1,
+        *,
+        enable_pickling: bool = False,
+    ) -> None:
+        super().__init__(parametrization=parametrization, budget=budget, num_workers=num_workers)
+        self._enable_pickling = enable_pickling
```

Comment:
how about we leave it as a public attribute? So that people can modify it easily on any instance. I'm not sure whether we should default to False (and then fail pickling) or True (and then risk using too much memory) :s

Reply:
I've changed it to a public function `enable_picking`.